Remove obsolete file breaking build with sphinx-3
No idea of what it should be replaced with.

--- doc/sphinxext/indexcontent.html~	2016-02-08 10:50:17.000000000 +0100
+++ doc/sphinxext/indexcontent.html	2020-12-03 14:11:49.679626660 +0100
@@ -1,4 +1,3 @@
-{% extends "defindex.html" %}
 {% block tables %}
 
 <table class="contentstable" align="center">
